mp/src/game/server/momentum/mom_system_saveloc.cpp

```diff
@@ -281,8 +319,24 @@ void CSaveLocSystem::LevelShutdownPreEntity()
             kvCPs->AddSubKey(kvCP);
         }
 
+        // Add startmarks
+        KeyValues *kvStartMarks = new KeyValues("startmarks");
+        for (int track = 0; track < MAX_TRACKS; track++)
+        {
+            char szTrackNum[10];
+            Q_snprintf(szTrackNum, sizeof(szTrackNum), "%09i", track);
+            KeyValues *kvStartMark = new KeyValues(szTrackNum);
+
+            // Save location of valid startmarks only
+            if (SavedLocation_t *startMark = pPlayer->GetStartMark(track))
```

Comment:
I think this check should probably be done first so that there aren't *always* MAX_TRACKS savelocs saved for every map.

You'd only want to save the info you know you want, which in this case are all of the valid track start marks. Most maps might not even have any and this current code would be writing a ton of empty, redundant info.

I'd do the null check and if it's null, continue on. Only create these keyvalues if it's a valid startmark.

Reply:
This check has now been moved above any unneeded computation.